Report on total memory size as well as counts

dgynn opened the pull request:
This adds the total memsize for allocated and retained objects. I formatted the output to show the memsize first and the count second in parenthesis. Let me know if there is a better way to be more clear what is what.
```
Total allocated 41181580 (221034)
Total retained 10410968 (13135)
```

SamSaffron commented:
We need units there, its a bit confusing just with number

maybe: 

Total allocated:123MB (221034 objects)

dgynn commented:
I added labels for just "bytes" and "objects". Since the size of the values can vary a lot I couldn't decide on a unit other than bytes that worked for very large and very small numbers.
```
Total allocated: 40 bytes (1 objects)
Total retained:  0 bytes (0 objects)
```
and
```
Total allocated: 120000000 bytes (3000000 objects)
Total retained:  40000000 bytes (1000000 objects)
```
I've also removed the RVALUE adjustment for Ruby 2.2's ObjectSpace memsize_of which no longer needs it.

SamSaffron commented:
looks good, nice catch on the RVALUE adjustment
